src/main/java/com/restapi/carMarket/service/CarService.java

```diff
@@ -20,68 +20,96 @@
     @Autowired
     CarDao carDao;
 
-    public void insert(Car car) {
-        if(isValid(car))
-            carDao.save(car);
-        else
-            throw new CarNotValidException();
-    }
-
-    private boolean isValid(Car car) {
-        boolean valid = true;
-        if(isValidPrice(car))
-            valid = false;
-        else if(isValidYear(car))
-            valid = false;
-        return valid;
-    }
-
-    private boolean isValidYear(Car car) {
-        return car.getYear()<1885 || car.getYear()> LocalDateTime.now().getYear();
-    }
-
-    private boolean isValidPrice(Car car) {
-        return car.getPrice()<1 || car.getPrice()>2000000000;
-    }
-
-    public List<Car> findAll() {
+    public List<Car> getAllCarsOnMarket() {
         return carDao.findAll();
     }
 
-    public Car findById(Long id) {
+    public Car getCarById(Long id) {
         Optional<Car> car = carDao.findById(id);
 
         if(car.isPresent())
             return car.get();
         else
-            throw new CarNotFoundException();
+            throw new CarNotFoundException("Car with id("+id+") is not found");
     }
 
-    public void delete(Car car) {
+    public void removeCarFromMarket(Car car) {
         if(carDao.exists(Example.of(car)))
             carDao.delete(car);
         else
-            throw new CarNotFoundException();
+            throw new CarNotFoundException("A Car with properties("+car.toString()+") is not found");
 
     }
 
-    public void deleteById(Long id) {
+    public void removeCarFromMarketById(Long id) {
         if(carDao.existsById(id))
             carDao.deleteById(id);
         else
-            throw new CarNotFoundException();
+            throw new CarNotFoundException("Car with id("+id+") is not found");
     }
 
-    public void update(Long id, Car car) {
-        if(isValid(car)){
-            Optional<Car> oldCar = carDao.findById(id);
+    public Car updateCarInformation(Long id, Car car) {
+        checkIfCarIsValid(car);
+        Optional<Car> oldCar = carDao.findById(id);
+        
+        if(oldCar.isPresent()) {
+            BeanUtils.copyProperties(oldCar.get(), car, "id");
+            return oldCar.get();
+        }
+        else{
+            car.setPostTime(LocalDateTime.now());
+            return carDao.save(car);
```

Comment:
Looks like you could just call `addCarToMarket()` here

Reply:
done